### ISSUE TITLE
BUG: 

### ISSUE THREAD

**Issue by novazembla** [10up/ElasticPress#2652]
**Describe the bug**
Hello, I'm running ElasticPress 4.0.0 on two different boxes (one is running Elasticsearch 7.9/PHP 8.0 the other 7.10/PHP 8.1) and experience that after index creation `post_date` and `post_date_gmt` are mapped as text fields and not date fields in the index. 

As I know they are set to be dates in `../mappings/7-0.php` I believe that this might be a bug. I cleared the index several times and tried to figure out if I could somehow introduce the problem but I could not find the reason that the mapping changes on index creation.  

Ad export of the mapping using `wp elasticpress get-mapping --index-name=...` shows the following field definitions. 

```
"post_date": {
  "type": "text",
  "fields": { "keyword": { "type": "keyword", "ignore_above": 256 } }
},
"post_date_gmt": {
  "type": "text",
  "fields": { "keyword": { "type": "keyword", "ignore_above": 256 } }
},
```

Question is do I do something wrong or is something wrong. Thank you!

**Steps to Reproduce**

1. Enable the plugin on a blank test Wordpress install
2. Sync the site with elastic search
3. Export the mapping via WP Cli and find that the fields are configured to be text. 

**Expected behavior**
`post_date` and `post_date_gmt` to be mapped as date, so all the filters and scoring methods work as they should. 

**Environment information**
 - Device: MacOs + Linux
 - WordPress version: 5.9.2
 - ElasticPress version: 4.0.0
 - Elasticsearch version: 7.9, 7.10
 - What ElasticPress features do you have currently active? PostSearch + Terms

**Comment by felipeelia** [comment]
Hi @novazembla,

I tried to reproduce the problem and I could not. Do you happen to have any custom code related to ElasticPress?

One thing you could do to determine how/if the mapping is changing before being sent to Elasticsearch is to use the [ep_post_mapping](https://github.com/10up/ElasticPress/blob/develop/includes/classes/Indexable/Post/Post.php#L385) filter with a code like:
```
add_filter(
	'ep_post_mapping',
	function( $post_mapping ) {
		error_log( var_export( $post_mapping, true ) );
		return $post_mapping;
	},
	1000
);
```

**Comment by novazembla** [comment]
Thanks for the quick reply. Adding this filter shows, that as expected these mappings are sent in. 

```
'post_date' => 
      array (
        'type' => 'date',
        'format' => 'yyyy-MM-dd HH:mm:ss',
      ),
      'post_date_gmt' => 
      array (
        'type' => 'date',
        'format' => 'yyyy-MM-dd HH:mm:ss',
      ),
```
But I think I found the culprit. It's the Terms feature (I only use Post Search and Terms). If I disable it the mapping is correct. It sends a second mapping that seems to wipe all the previously indexed posts. 

Here is the log of the "Deleta All data an fresh sync"
```
1. Deleting data…
2. Indexing posts...
3. Mapping sent
4. Processed posts 0 - 350 of 1175. Last Object ID: 113664
5. Processed posts 350 - 700 of 1175. Last Object ID: 113312
6. Processed posts 700 - 1050 of 1175. Last Object ID: 24581
7. Processed posts 1050 - 1175 of 1175. Last Object ID: 255
8. Number of posts indexed: 1175
9. Indexing terms...
10. Mapping sent
11. Processed terms 0 - 350 of 496. Last Object ID: 2771
12. Processed terms 350 - 700 of 496. Last Object ID: 2771
13. Number of terms indexed: 700
14. Sync complete
===============================
```
But if I go and check the index I only see 350 items. As visible on this screenshot. 
<img width="168" alt="image" src="https://user-images.githubusercontent.com/279480/158170577-b343ab95-caff-4539-906b-5ddc6bd387b7.png">

And the created mapping is rather short. (See attachment: mappingAfterDeleteAllAndSync.json)

I already wondered why I always had to run a second manual sync to have actually all posts synced. When I do this the posts are there but the mapping is wrong (See attachment: mappingAfterManualSync.json). I guess ElasticSearch infers the mapping based on the submitted data but as post_date is submitted as string is becomes a test field.  

[mappings.zip](https://github.com/10up/ElasticPress/files/8244714/mappings.zip)

TLDR; If I disable the Terms Feature the mapping and syncing works as it should. If I enable terms, I've to manually trigger a sync after a delete and sync to have to posts in the index and the mappings are wrong. 

Could this be a ElasticSearch setting that is configured wrong? Bear in mind that I'm testing on two machines with EP4.0.0 and EL7.9 and EL7.10. 

Thank you!

**Comment by novazembla** [comment]
@felipeelia that makes sense. As yes I do hard code the index name making use of the `ep_index_name` filter as I'm also using external scripts to query elasticsearch directly and it's more convenient to rely on shared index names. 

I've changed that filter callback to allow for multiple indexes. Now all works as it should. 

Thanks for your help!
